src/libstd/keyword_docs.rs

```diff
@@ -283,7 +305,7 @@ mod fn_keyword { }
 
 #[doc(keyword = "for")]
 //
-/// The `for` keyword.
+/// Iterate over a series of values with [`in`], or implement a trait for a type with [`impl`].
```

Comment:
This is missing HRTB `for<'a>`

Reply:
That's true, although `for<'a>` is mentioned in the next paragraph. I tried to avoid simply enumerating all of the uses of a keyword in the summary and instead giving the reader a sense of its meaning in rust programs. That's why `impl` and `const` are so vague. I wasn't able to do this with `for`, since looping and `impl Trait for T` are fundamentally different.

Reply:
The hope is that even if more uses of a keyword are added to the language, its general meaning will stay the same. For example, `impl Trait` describes what functionality a return type implements.

Reply:
```suggestion
/// Iteration with [`in`], trait implementation with [`impl`], or [higher-ranked trait bounds] (`for<'a>`).
```

After taking a break and coming back, I think I'm in favor of including HRTBs in the summary line. It seems conceivable that someone would search through the `std` docs when they first encounter `for<'a>`, and seeing an example of it in the search results would be useful.